mlflow/utils/model_utils.py

```diff
@@ -428,3 +429,70 @@ def _validate_pyfunc_model_config(model_config):
             "valid file path or of type ``dict`` with string keys.",
             error_code=INVALID_PARAMETER_VALUE,
         )
+
+
+RECORD_ENV_VARS = {
+    # api key related
+    "API_KEY",
+    # databricks auth related
+    "DATABRICKS_HOST",
+    "DATABRICKS_USERNAME",
+    "DATABRICKS_PASSWORD",
+    "DATABRICKS_TOKEN",
+    "DATABRICKS_INSECURE",
+    "DATABRICKS_CLIENT_ID",
+    "DATABRICKS_CLIENT_SECRET",
+    "_DATABRICKS_WORKSPACE_HOST",
+    "_DATABRICKS_WORKSPACE_ID",
+    # dbconnect related
+    "SPARK_REMOTE",
+    "SPARK_LOCAL_REMOTE",
+}
+
+
+@contextlib.contextmanager
+def env_var_tracker():
```

Comment:
Does this capture environment variables accessed by mlflow?

Reply:
Yes, but `MLFLOW_xxx` is not in the allowlist